### PR TITLE
[export] Update schema version

### PR DESCRIPTION
**Pull request by angelayi**
Since pytorch 2.1 release we've made some BC breaking changes to the serialized schema. We should update it in time for the 2.2 release. Some of the changes include:

* https://github.com/pytorch/pytorch/pull/114371 - custom class objects / pybinded objects are no longer saved directly to the `ExportedProgram` structure. Instead, the name is serialized inside of the program, and the actual bytes are stored. in a separate location from the exported program, allowing it to be saved to a different location.
* https://github.com/pytorch/pytorch/pull/111204 - `GraphSignature` structure changed and `call_spec` is removed from the `GraphModule` schema
* https://github.com/pytorch/pytorch/pull/111407 - `loss_outout` -> `loss_output`
* https://github.com/pytorch/pytorch/pull/113075 - `example_inputs` removed from the `ExportedProgram` structure (this originally did not store anything), `dialect` added to the `ExportedProgram` structure.
* https://github.com/pytorch/pytorch/pull/113689 - tensor constants are now lifted as inputs to the graph, and their locations are stored in the `GraphSignature`
* https://github.com/pytorch/pytorch/pull/114172 - removed `equality_constraints` and added a `SymExprHint` for all symbolic expressions.

### CODE REVIEW

File: torch/_export/serde/schema.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 
 # NOTE: Please update this value if any modifications are made to the schema
-SCHEMA_VERSION = 1
+SCHEMA_VERSION = 2
```
Comment (tugsbayasgalan) [inline torch/_export/serde/schema.py:10]: not blocking: should we add comment here describing what is changed? In general, whenever there is the version update, there should be some sort of doc or tracker?

Comment (angelayi) [reply torch/_export/serde/schema.py:10]: ooh. I'm not sure what should be done during a version bump. I can add it to this PR and make sure it's in the release notes?

Comment (tugsbayasgalan) [reply torch/_export/serde/schema.py:10]: What if we paste it in the code as comment as well?

**Comment by zhxchen17** [comment]
@angelayi considering export is still before alpha release, should we keep it 0? I feel it's rather confusing to see this to be bumped right now.

**Comment by angelayi** [comment]
@zhxchen17 I think export is *at* alpha release, so it's up to us how we want to do versioning. From a user's pov, the schema changed between 2.1 and 2.2 release, which is why I think we should bump the schema to show this change. 
It seems your opinion is that we should bump the version when export gets to beta/actual release.
And I think @gmagogsfm 's opinion is that we should start maintaining BC now.

**Comment by zhxchen17** [comment]
> @zhxchen17 I think export is _at_ alpha release, so it's up to us how we want to do versioning. From a user's pov, the schema changed between 2.1 and 2.2 release, which is why I think we should bump the schema to show this change. It seems your opinion is that we should bump the version when export gets to beta/actual release. And I think @gmagogsfm 's opinion is that we should start maintaining BC now, and add back in logic that allows loading the schema from 2.1.

well if there's no major user of the schema in real life, I'd rather make the breaking change than taking the cost to maintain it. 

Also I'm not sure we're at alpha release really:

Screenshot 2023-12-13 at 12 52 50 PM

**Comment by angelayi** [comment]
Is alpha != prototype?

**Comment by gmagogsfm** [comment]
Alpha == Prototype, but Alpha is still open sourced, so we need more guarantee than private prototype. It is debatable whether we want to take on the responsibility of maintaining BC-safe loader since it involves additional eng work. But I think it is a bare minimum for us to update schema version per PyTorch release since we have breaking changes. @zhxchen17 thoughts?

**Review by gmagogsfm — APPROVED** [review]

**Pull request closed without merging**